includes/Modules/Analytics_4.php

```diff
@@ -1314,27 +1318,27 @@ private function get_custom_dimensions_data() {
 		 * @param array $allowed_post_types The array of allowed post types.
 		 */
 		$allowed_post_types = apply_filters( 'custom_dimension_valid_post_types', array( 'post' ) );
-		if ( ! is_singular( $allowed_post_types ) ) {
-			return array();
-		}
 
 		$data = array();
 		$post = get_queried_object();
 
-		foreach ( $settings['availableCustomDimensions'] as $custom_dimension ) {
-			switch ( $custom_dimension ) {
-				case 'googlesitekit_post_type':
-					$data[ $custom_dimension ] = $post->post_type;
-					break;
-				case 'googlesitekit_post_author':
-					$data[ $custom_dimension ] = $post->post_author;
-					break;
-				case 'googlesitekit_post_categories':
-					$data[ $custom_dimension ] = implode( ',', wp_get_post_categories( $post->ID, array( 'fields' => 'ids' ) ) );
-					break;
-				case 'googlesitekit_post_date':
-					$data[ $custom_dimension ] = get_the_date( 'Ymd', $post );
-					break;
+		if ( in_array( 'googlesitekit_post_type', $settings['availableCustomDimensions'] ) ) {
```

Comment:
Hi @eugene-manuilov, the logic needs a further tweak - as per @nfmohit's most recent [comment](https://github.com/google/site-kit-wp/issues/7600#issuecomment-1767229540), we should only track `googlesitekit_post_type` if `is_singular()` is true for the current post. The remaining custom dimensions should only be tracked if `is_singular()` is true _and_ the current post type is the allowed post type (i.e. `post` by default).

Reply:
Also, this usage of `in_array()` needs its third argument specified to satisfy PHP Lint.